File: 1004 max-consecutive-ones-iii/solution.cs
public class Solution {
    public int LongestOnes(int[] nums, int k) {
        int l=0,m=0;
        for(int r=0;r<nums.Length;r++) {
            if(nums[r]==0) k--;
            while(k<0) {
                if(nums[l]==0) k++;
                l++;
            }
            m=Math.Max(m,r-l+1);
        }
        return m;
    }
}

File: 1004 max-consecutive-ones-iii/solution.rs
impl Solution {
    pub fn longest_ones(nums: Vec<i32>, k: i32) -> i32 {
        let mut l=0;
        let mut m=0;
        let mut k=k as i32;

        for r in 0..nums.len() {
            if nums[r]==0 {
                k-=1;
            }
            while k<0 {
                if nums[l]==0 {
                    k+=1;
                }
                l+=1;
            }
            m=m.max((r-l+1) as i32);
        }
        m
    }
}
